src/main/java/duke/ui/Parser.java

```diff
@@ -0,0 +1,129 @@
+package duke.ui;
+
+import duke.Tasks.Storage;
+
+import java.util.Scanner;
+/**
+ * The main driver class of the chatbot, called by notGPT
+ */
+public class Parser {
+    /**
+     * Runs the entire program, reading user input and deciding what to do
+     *
+     * @param scanner system input from command line
+     */
+    public static void parse(Scanner scanner) {
+        Storage storage = new Storage();
+        String command;
+        String text ="";
+        String input;
+        do {
+            input = scanner.nextLine();
+            String[] parts = input.split("\\s+", 2);
+            command = parts[0].toLowerCase();
+            if (parts.length > 1 && !parts[1].trim().isEmpty()) {
+                text = parts[1];
+            }
+            switch (command) {
+            case "list":
```

Comment:
I like how you didn't indent the `case` statements, as per the coding standards. Good job!